### ISSUE TITLE
tsx and jsx at same time supported?

### ISSUE THREAD

**Issue by torian257x** [raicem/laravel-mix-react-typescript-extension#1]
btw I love this plugin, I am using it for most of my projects now! works great, and stripping ts is the right decision as well imho. I have very little bugs.

Now to my problem:

I started a project in plain jsx

but now I want to move to tsx. Is it possible to have tsx and jsx files at the same time without having to port the ~50 files I created already?

as all typescript is stripped anyhow, this should somehow work, no?

But I get 

```
ERROR in ./src/myapp-react-app.jsx 83:6
Module parse failed: Unexpected token (83:6)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| if (root_el) {
|   render(
>       <>
|         <myappCitationSlider/>
|       </>
 @ multi ./src/myapp-react-app.jsx /dist/myapp-react-app[0]
```

**Comment by raicem** [comment]
Thank you! Honestly, I have no idea atm but I'm looking at this line:

https://github.com/raicem/laravel-mix-react-typescript-extension/blob/master/index.js#L38

Only `ts` and `tsx` files are sent to the `babel-loader`. That may be the problem. I have no idea what happens if I add `js` and `jsx` there.

I'll give this one a go and report back.

**Comment by torian257x** [comment]
@raicem how did it go?

**Comment by raicem** [comment]
Sorry it took me a bit long to get back. I think I had success. Here's what I did.

Installed a brand new Laravel project, setup the react preset using the new `laravel/ui` package. Then I installed this extension and I tried to compile React files with it. 

It gave me an error similar like yours. So I think I reproduced it. I then applied my fix to it and it was able to compile `js` files as well. I have committed and tagged my and it should be updated at npm as well.

Release: https://github.com/raicem/laravel-mix-react-typescript-extension/releases/tag/1.0.1
NPM: https://www.npmjs.com/package/laravel-mix-react-typescript-extension

**Comment by torian257x** [comment]
@raicem I think you broke your project eventually... problem is, that somehow it seems to have matched scss files, e.g. it tried somehow to build scss files with this webpack config:

```
const mix = require('laravel-mix');
require("laravel-mix-react-typescript-extension");

mix.reactTypeScript('src/myapp-react-app.jsx',  'dist/');
mix.sass('../assets/scss/myapp.scss', '../assets/css/myapp.css');
mix.copy('dist/*', '../assets/js/react-dist/');
```


now I played around with your index.js and it seems that this works, and it is as well better readable

`test: /(\.js)|(\.jsx)|(\.ts)|(\.tsx)$/,`

because, afaik:

`abc|d`

will match
`abd`


and you had a bunch of  unescaped dots, basically matching everything?


the error I got was btw this:


```
ERROR in ../assets/scss/myapp.scss (./node_modules/css-loader??ref--8-2!./node_modules/postcss-loader/src??postcss0!./node_modules/resolve-url-loader??ref--8-4!./node_modules/sass-loader/dist/cjs.js??ref--8-5!./node_modules/babel-loader/lib??ref--14-0!../assets/scss/myapp.scss)
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /home/myuser/IdeaProjects/myapp/src_dev/assets/scss/myapp.scss: Unexpected token, expected "(" (1:8)

> 1 | @import 'variables';
    |         ^
  2 | @import 'mixins/mixins';
  3 | @import 'utility/_utility.scss';
  4 | @import 'layout/_layout.scss';

```

**Comment by torian257x** [comment]
maybe even a `$` in each ( ) ? because not sure what (a)|(b)$ matches honestly... not entirely sure. I would assume a$ or b$ 
but (a$)|(b$) is pretty clear what it matches

**Comment by raicem** [comment]
Yeah I copied and pasted the previous config without giving a much thought. I didn't have any scss compilation in my webpack config so I must have missed that. Thanks for the help.

I have updated the config as per your regex:

`test: /(\.js)|(\.jsx)|(\.ts)|(\.tsx)$/,`

This seems to be doing the trick. I mean... I hope. It's released as 1.0.2 now.

**Comment by torian257x** [comment]
@raicem I used it for several days now and works great! excellent!
